### DISCUSSION TITLE
Is it possible to use a keyboard/mouse to interact with an app running on a Raspberry Pi via DRM?

### DISCUSSION THREAD

**Discussion by s-g-stavely** [AvaloniaUI/Avalonia#11748] [Q&A] [answered]
Hi, I followed this guide to run a simple app on a Raspberry Pi with Raspberry Pi OS Lite installed. https://docs.avaloniaui.net/guides/deep-dives/running-on-raspbian-lite-via-drm

The app looks fine, but keyboard and mouse input don't seem to get picked up by the app. Hitting Tab doesn't select different controls, and typing doesn't do anything. All I can do is hit Ctrl-C to exit the app. When running on Windows, input works fine.

I'm using Avalonia 11.0.0-rc1.

I'd like to be able to type into a text box, and ideally to use the mouse to select things.

Is there a way to do this? Or is this just some limitation of running using DRM, and I should just write a normal Linux app to run in the full Raspberry Pi OS instead of the Lite one?

**Comment by jp2masa** [comment, marked as answer]
#2920?

**Reply by s-g-stavely** [reply]
That looks like the same issue, although I don't understand the relationship between fbdev and DRM well enough to say for sure. I'll assume this is not possible for now. Thanks!

**Comment by kekekeks** [comment]
The framebuffer platform is currently focused on devices with touch input. For mouse support we'll need to implement a cursor overlay (preferably as a DRM overlay), for keyboard we'll need to integrate libxkbcommon to utilize layouts. 

There are currently no immediate plans for implementing that.

**Reply by s-g-stavely** [reply]
Gotcha, thanks!

**Reply by ucar19** [reply]
Is there any news about this?

**Reply by MikeCodesDotNET** [reply]
If this is something you require, we could discuss you sponsoring this work. Feel free to contact us at sales@avaloniaui.net
